### PR TITLE
feat: bump `cosmos-sdk` to `v0.47.1`

### PR DESCRIPTION
**Pull request by Pantani**

### CODE REVIEW

File: proto/buf.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: v1
+name: github.com/Pantani/earth
```
Comment (jeronimoalbi) [inline proto/buf.yaml:2]: Shouldn't the name be something like `buf.build/ignite/modules`?

File: x/mint/module_simulation.go [hunk 2]
```diff
@@ -19,13 +17,13 @@ func (AppModule) GenerateGenesisState(simState *module.SimulationState) {
 }
 
 // ProposalContents doesn't return any content functions for governance proposals.
-func (AppModule) ProposalContents(simState module.SimulationState) []simtypes.WeightedProposalContent {
+func (AppModule) ProposalContents(_ module.SimulationState) []simtypes.WeightedProposalContent { //nolint:staticcheck
```
Comment (jeronimoalbi) [inline x/mint/module_simulation.go:20]: I believe `ProposalContents` and `ProposalMsgs` could be removed if no proposal is being tested

File: cmd/cosmosapp.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// CosmosApp implements the common methods for a Cosmos SDK-based application
+// specific blockchain.
+type CosmosApp interface {
```
Comment (jeronimoalbi) [inline cmd/cosmosapp.go:12]: Out of curiosity, why is this interface defined?

File: app/simulation_test.go [hunk 4]
```diff
@@ -57,21 +60,30 @@ func interBlockCacheOpt() func(*baseapp.BaseApp) {
...
 	b.Cleanup(func() {
-		db.Close()
-		err = os.RemoveAll(dir)
-		require.NoError(b, err)
+		require.NoError(b, db.Close())
+		require.NoError(b, os.RemoveAll(dir))
 	})
+	appOptions := make(simtestutil.AppOptionsMap, 0)
+	appOptions[flags.FlagHome] = app.DefaultNodeHome
+	appOptions[server.FlagInvCheckPeriod] = simcli.FlagPeriodValue
```
Comment (jeronimoalbi) [inline app/simulation_test.go:83]: I think the `appOptions` is not being used, if so it could be removed

File: cmd/cosmosapp.go (hunk 3 above)
Comment (Pantani) [reply cmd/cosmosapp.go:12]: I kept it by mistake. Removed

**Review by jeronimoalbi — APPROVED** [review]

**Pull request merged**